### PR TITLE
[DO NOT MERGE] Upgrade auth0 to v4

### PR DESCRIPTION
**Pull request by frederikprijck**
### 🔧 Changes

Just a temporary PR to show what migration to v4 of auth0 would look like.

### TODO

- [X] Add pagination overloads to `clients.getAll` (have updated the spec)
- [X] Figure out why we had to add extra `PATCH` `/clients` and `DELETE` `/client-grants` to the test fixtures (also happening for me on `master`)

### 🔬 Testing

- Unit tests work
- ~~E2e tests work locally, but needs a change in node-auth0 regarding `client_id` (it currently takes an `id` instead of `client_id`, which we want to look into a bit more.~~

### 📝 Checklist

- [ ] All new/changed/fixed functionality is covered by tests (or N/A)
- [ ] I have added documentation for all new/changed functionality (or N/A)

### CODE REVIEW

File: src/tools/auth0/handlers/tenant.ts [hunk 1]
```diff
@@ -66,19 +66,20 @@ export default class TenantHandler extends DefaultHandler {
     // Do nothing if not set
     if (!tenant) return;
 
-    const updatedTenant: Partial<Tenant> = {
-      ...tenant,
+    const updatedTenant: TenantSettingsUpdate = {
+      // TODO: What's wrong with the enums?
```
Comment (adamjmcgrath) [inline src/tools/auth0/handlers/tenant.ts:70]: Think it's this https://github.com/auth0/api2/blob/8925c49da64a117b19dbedb37e47e1365198a37c/api/tenants/schemas/tenant_settings_update_properties.js#L105

File: src/tools/auth0/handlers/triggers.ts [hunk 2]
```diff
@@ -58,14 +58,15 @@ export default class TriggersHandler extends DefaultHandler {
...
       const res = await this.client.actions.getAllTriggers();
-      const triggers: string[] = _(res.triggers).map('id').uniq().value();
+      const triggers: string[] = _(res.data.triggers).map('id').uniq().value();
 
       for (let i = 0; i < triggers.length; i++) {
         const triggerId = triggers[i];
-        const { bindings } = await this.client.actions.getTriggerBindings({
-          trigger_id: triggerId,
+        const { data } = await this.client.actions.getTriggerBindings({
+          triggerId: triggerId,
```
Comment (adamjmcgrath) [inline src/tools/auth0/handlers/triggers.ts:66]: Another one we might want to look at

File: src/context/directory/index.ts [hunk 3]
```diff
@@ -21,7 +21,7 @@ export default class DirectoryContext {
   assets: Assets;
   disableKeywordReplacement: boolean;
 
-  constructor(config: Config, mgmtClient: Auth0APIClient) {
+  constructor(config: Config, mgmtClient: BaseAuth0APIClient) {
```
Comment (adamjmcgrath) [inline src/context/directory/index.ts:24]: `ManagementClient`?

File: src/tools/auth0/handlers/organizations.ts [hunk 4]
```diff
@@ -135,7 +136,7 @@ export default class OrganizationsHandler extends DefaultHandler {
       connectionsToUpdate.map((conn) =>
         this.client.organizations
           .updateEnabledConnection(
-            { connection_id: conn.connection_id, ...params },
+            { connectionId: conn.connection_id, ...params },
```
[the remark this answers is not in the document]
Comment (frederikprijck) [reply src/tools/auth0/handlers/organizations.ts:139]: Lets get a list of all paramaters and indicate whether or not is shows up in the response, would be useful to also include if it shows up in a different casing variant.

File: src/tools/auth0/handlers/emailProvider.ts [hunk 5]
```diff
@@ -16,7 +17,8 @@ export default class EmailProviderHandler extends DefaultHandler {
 
   async getType(): Promise<Asset> {
     try {
-      return await this.client.emailProvider.get({ include_fields: true, fields: defaultFields });
+      const { data  } = await this.client.emails.get({ include_fields: true, fields: defaultFields.join(',') });
```
Comment (frederikprijck) [inline src/tools/auth0/handlers/emailProvider.ts:20]: @adamjmcgrath This is a breaking change we want to call out.

Comment (adamjmcgrath) [reply src/tools/auth0/handlers/emailProvider.ts:20]: https://github.com/auth0/node-auth0/blob/beta/v4_MIGRATION_GUIDE.md#response-change

Comment (adamjmcgrath) [reply src/tools/auth0/handlers/emailProvider.ts:20]: Ah, sorry - this is from our chat about `fields` - ignore me....

**Comment by adamjmcgrath** [comment]
Noticed one issue when running e2e tests against a live tenant (using `wild` mode) and raised SDK-4548

**Comment by willvedd** [comment]
Keeping this PR closed but restoring the original branch.

**Pull request closed without merging**